mlir/include/mlir/Translation/GpuModuleToRocdir.h

```diff
@@ -0,0 +1,26 @@
+//===- GpuModuleToRocdlir.h - ROCDL in GPU modules to LLVM IR ----*- C++
+//-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This provides registration calls for ROCDL dialect to LLVM IR translation in
+// GPU modules.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_TRANSLATION_GPUMODULETOROCDLIR_H
+#define MLIR_TRANSLATION_GPUMODULETOROCDLIR_H
+
+namespace mlir {
+namespace miopen {
+/// A translation that converts LLVM Dialect within a GPU module
+/// to LLVM IR
+void registerGpuModuleToROCDLIRTranslation();
```

Comment:
Okay there it is. Why do you need to move and rename tho? Did you internalize this translation?

Reply:
> Due to an initial circular dependency issue

What happened with circular dependency?

Reply:
GPUOps depended on the translation which depended on GPUOps

The actual problem was a case of the dependencies for SerializeToHsaco being linked to MlirGpuOps instead of MlirGpuTransforms, so it was unrelated, but I decided to move this translation internally.

(It's internal now because it's weirdly us-specific, mainly used for testing, and something of a workaround for a previous change upstream. I have plans to replace that translation with a `gpu-modules-only` pass followed by `-mlir-to-llvmir` or something to that effect)